config/foreman-proxy-content.migrations/200630000120-disable-pulp-2-el8.rb

```diff
@@ -0,0 +1,4 @@
+if facts[:os][:release][:major] == '8' && facts[:os][:family] == 'RedHat'
+  answers['foreman_proxy::plugin::pulp']['enabled'] = false
```

Comment:
Let's be defensive and first check that `answers['foreman_proxy::plugin::pulp']` is a hash. Otherwise you can get nasty errors. Might also be good to handle `true`.

Reply:
What is the case where it is not a hash?

Reply:
In the future we may change the answer file to just `foreman_proxy::plugin::pulp: true`. Users can also have set it to `false` (though that's unlikely). It's mostly that I'm allergic to code that assumes a value and crashes with an exception on other legal values.

Reply:
Be nice if there was a way to enforce this without it being still possible for the user to set a bad option and get themselves in trouble. I liked the way the hiera heirarchy works but the answers file ultimately overrides that. Being able to block parameters would also be nice.

Reply:
I too want to be more flexible in that. I thought I had a way to override the module defaults at the scenario level but misunderstood how Kafo does things today. Still on my list to fix.

> Being able to block parameters would also be nice.

Blocking values is possible with a `pre_commit` hooks. Being able to prevent parameters from being shown as parameters is currently not possible but also on my wish list.

Combining those two (scenario level defaults + hidden parameters) would be a powerful tool in reducing parameter counts.